src/scenarios/helloandroid/pre.py

```diff
@@ -16,27 +16,29 @@
 parser.add_argument('--unzip', help='Unzip APK and report extracted tree', action='store_true', default=False)
 args = parser.parse_args()
 
-apkname = 'HelloAndroid.apk'
-apknamezip = '%s.zip' % (apkname)
-if not os.path.exists(apkname):
-    print('Cannot find %s' % (apkname))
-    exit(-1)
-if args.unzip:
-    if not os.path.exists(apknamezip):
-        copyfile(apkname, apknamezip)
-
-    with ZipFile(apknamezip) as zip:
-        zip.extractall(os.path.join('.', PUBDIR))
-
-    assets_dir = os.path.join(PUBDIR, 'assets')
-    assets_zip = os.path.join(assets_dir, 'assets.zip')
-    with ZipFile(assets_zip) as zip:
-        zip.extractall(assets_dir)
-
-    os.remove(assets_zip)
-else:
+if not os.path.exists(PUBDIR):
     os.mkdir(PUBDIR)
-    copyfile(apkname, os.path.join(PUBDIR, apkname))
+apknames = ['HelloAndroid.apk', 'HelloAndroidWithDiag.apk']
+for apkname in apknames:
+    apknamezip = '%s.zip' % (apkname)
+    if not os.path.exists(apkname):
+        getLogger().error('Cannot find %s' % (apkname))
+        exit(-1)
```

Comment:
Do we want to fail if an APK is not found, or do we want to rely on the calling of the test method at test time to error if the apk is not found.

Reply:
I think that we should fail at the first time that we discover that there is an error. So in that case, it should be when we realize that we do not have the APKs in the pre step.